### PR TITLE
Update spanner queries

### PR DESCRIPTION
**Pull request by n-h-diaz**
* Updates node queries to use only GQL. This helps improve performance for querying with filters
* Removes extra join in observation with contained in place query. After speaking with Mingtian, he recommends adding observation to the graph to remove additional joins. This will require updating the graph schema, and there are some implications for write throughput, but we can potentially explore this (can discuss more later)
* Updates some search tests (it's unclear if this is just flaky, so if it starts to fail again, we can consider updating the test to support potentially inconsistent ordering)

### CODE REVIEW

File: internal/server/spanner/query.go [hunk 1]
```diff
@@ -58,193 +58,144 @@ var statements = struct {
...
-		DISTINCT subject_id,
-		predicate
-	FROM 
-		Edge
-	WHERE
-		subject_id IN UNNEST(@ids)
-	ORDER BY
-		subject_id,
-		predicate
+		GRAPH DCGraph MATCH -[e:Edge
```
Comment (keyurva) [inline internal/server/spanner/query.go:61]: Do we have any numbers (or a sense for) on the performance improvements with GQL?

**Review by keyurva — APPROVED** [review]
Thanks - excited about the perf improvements!

Let's chat about observations in the graph.

File: internal/server/spanner/golden/query/search_nodes_with_type.json [hunk 2]
```diff
@@ -349,6 +349,13 @@
       "StatisticalVariable"
     ]
   },
+  {
+    "SubjectID": "oecd/IDD_INCACTOTAL_TOT_CURRENT_METH2011",
```
[the remark this answers is not in the document]
Comment (n-h-diaz) [reply internal/server/spanner/golden/query/search_nodes_with_type.json:353]: He said he thinks it's consistent but will confirm and get back to us. He also suggested doing a hybrid search (see the question notes doc)

File: internal/server/spanner/query.go (hunk 1 above)
Comment (n-h-diaz) [reply internal/server/spanner/query.go:61]: It should speed up queries with filters significantly

**Pull request merged**